imap_processing/glows/l1/glows_l1a.py

```diff
@@ -0,0 +1,405 @@
+) -> xr.Dataset:
+    """
+    Generate a dataset for GLOWS L1A histogram data CDF files.
+
+    Parameters
+    ----------
+    hist_l1a_list : list[HistogramL1A]
+        List of HistogramL1A objects for a given day
+    data_version : str
+        Data version for CDF filename, in the format "vXXX"
+
+    Returns
+    -------
+    output : xr.Dataset
+        Dataset containing the GLOWS L1A histogram CDF output
+    """
+    time_data = np.zeros(len(hist_l1a_list), dtype="datetime64[ns]")
+    # TODO Add daily average of histogram counts
+    # TODO compute average temperature etc
+    # Data in lists, for each of the 25 time varying datapoints in HistogramL1A
+
+    hist_data = np.zeros((len(hist_l1a_list), 3600))
+
+    # TODO: add missing attributes
+    support_data = {
+        "flight_software_version": [],
+        # TODO: should this be a global file attribute?
+        # "ground_software_version": [],
+        # "pkts_file_name": [],
+        "seq_count_in_pkts_file": [],
+        "last_spin_id": [],
+        "flags_set_onboard": [],
+        "is_generated_on_ground": [],
+        "number_of_spins_per_block": [],
+        "number_of_bins_per_histogram": [],
+        "number_of_events": [],
+        "filter_temperature_average": [],
+        "filter_temperature_variance": [],
+        "hv_voltage_average": [],
+        "hv_voltage_variance": [],
+        "spin_period_average": [],
+        "spin_period_variance": [],
+        "pulse_length_average": [],
+        "pulse_length_variance": [],
+    }
+    time_metadata = {
+        "imap_start_time": [],
+        "imap_time_offset": [],
+        "glows_start_time": [],
+        "glows_time_offset": [],
+    }
+
+    for index, hist in enumerate(hist_l1a_list):
+        # TODO: Should this be MET?
+        epoch_time = calc_start_time(hist.imap_start_time.to_seconds())
+        hist_data[index] = hist.histograms
+
+        support_data["flags_set_onboard"].append(hist.flags["flags_set_onboard"])
+        support_data["is_generated_on_ground"].append(
+            int(hist.flags["is_generated_on_ground"])
+        )
+
+        for key in support_data.keys():
+            if key not in ["flags_set_onboard", "is_generated_on_ground"]:
+                support_data[key].append(hist.__getattribute__(key))
+        for key in time_metadata.keys():
+            time_metadata[key].append(hist.__getattribute__(key).to_seconds())
+        time_data[index] = epoch_time
+
+    epoch_time = xr.DataArray(
+        time_data,
+        name="epoch",
+        dims=["epoch"],
+        attrs=ConstantCoordinates.EPOCH,
+    )
+
+    bins = xr.DataArray(
+        np.arange(3600),  # TODO: Is it always 3600 bins?
+        name="bins",
+        dims=["bins"],
+        attrs=glows_cdf_attrs.bins_attrs.output(),
+    )
+
+    hist = xr.DataArray(
+        hist_data,
+        name="histograms",
+        dims=["epoch", "bins"],
+        coords={"epoch": epoch_time, "bins": bins},
```

Comment:
I don't think you need to pass in the `coords` directly to the DataArrays if you specify the coords at the Dataset level. As long as the dims are pointing to a coordinate array in the dataset (which you have done below) it should pick up that mapping.

Reply:
Agree. If you move line 371 up above these, then you won't need these coordinates.